### PR TITLE
Preventing UCs with same IDs to be duplicated

### PR DESCRIPTION
**Pull request by Darguima**
UMinho subjects page repeat the master subjects for different master options.
With this update if that ID already exists, it will not be joined to subjects list.

### CODE REVIEW

File: scraper/modules/subjects_scraper.py [hunk 1]
```diff
@@ -234,18 +236,23 @@ def scraper(driver: WebDriver, course_name: str, short_names, master: bool = Fal
                     f"\t\t\033[93m\033[1mWARNING:\033[0m Was not possible to find out \033[4m{subject_name}\033[0m's subject ID. Using 0 as default.")
             # =====================
 
-            if subject_id in short_names.keys():
-                short_name = short_names[subject_id]["short_name"]
+            if str(subject_id) in short_names.keys():
+                short_name = short_names[str(subject_id)]["short_name"]
```
Comment (diogogmatos) [inline scraper/modules/subjects_scraper.py:240]: Can you also display a warning on the console when this case happens? That way, whoever is running the scraper will know that there is a subject with a short name that needs to be replaced.

**Review by diogogmatos — COMMENTED** [review]
Other than what I commented, everything seems okay! Did you test this code?

File: scraper/modules/subjects_scraper.py (hunk 1 above)
Comment (Darguima) [reply scraper/modules/subjects_scraper.py:240]: This is not for prevent repeated short names. This is to get the short names from a previous file that calendarium initially used, from a commit before the scraper, trying to preserve the old short names.

I just did some small improvements with types.

That fix that you want will be done on other PR.

Comment (Darguima) [reply scraper/modules/subjects_scraper.py:240]: This is not for prevent repeated short names. This is to get the short names from a previous file that calendarium initially used, from a commit that exists before the scraper, trying to preserve the old short names.

I just did some small improvements with types.

That fix that you want will be done on other PR.

**Comment by Darguima** [comment]
> Other than what I commented, everything seems okay! Did you test this code?

Yes I did. All the repeated subjects at 4 year disappeared.

**Review by diogogmatos — APPROVED** [review]
Good job 🚀

**Pull request merged**